pkg/frontend/fixetcd.go

```diff
@@ -0,0 +1,658 @@
+	}
+	// Cluster Role isn't scoped to a namespace
+	clusterRole.SetAPIVersion("rbac.authorization.k8s.io/v1")
+	clusterRole.SetKind("ClusterRole")
+	clusterRole.SetName(usersAccount)
+	clusterRole.SetClusterName(cluster)
+
+	return clusterRole
+}
+
+func newClusterRoleBinding(name, cluster string) *unstructured.Unstructured {
+	crb := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"roleRef": map[string]interface{}{
+				"kind":      "ClusterRole",
+				"name":      kubeServiceAccount,
+				"apiGroups": "",
+			},
+			"subjects": []rbacv1.Subject{
+				{
+					Kind:      "ServiceAccount",
+					Name:      name,
+					Namespace: namespaceEtcds,
+				},
+			},
+		},
+	}
+	crb.SetAPIVersion("rbac.authorization.k8s.io/v1")
+	crb.SetKind("ClusterRoleBinding")
+	crb.SetName(name)
+	crb.SetClusterName(cluster)
+
+	return crb
+}
+
+func newSecurityContextConstraint(name, cluster, usersAccount string) *unstructured.Unstructured {
+	scc := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"groups":                   []string{},
+			"users":                    []string{usersAccount},
+			"allowPrivilegedContainer": true,
+			"allowPrivilegeEscalation": to.BoolPtr(true),
+			"allowedCapabilities":      []corev1.Capability{"*"},
+			"runAsUser": map[string]securityv1.RunAsUserStrategyType{
+				"type": securityv1.RunAsUserStrategyRunAsAny,
+			},
+			"seLinuxContext": map[string]securityv1.SELinuxContextStrategyType{
+				"type": securityv1.SELinuxStrategyRunAsAny,
+			},
+		},
+	}
+	scc.SetAPIVersion("security.openshift.io/v1")
+	scc.SetKind("SecurityContextConstraints")
+	scc.SetName(name)
+	scc.SetClusterName(cluster)
+
+	return scc
+}
+
+// createPrivilegedServiceAccount creates the following objects and returns a cleanup function to delete them all after use
+//
+// - ServiceAccount
+//
+// - ClusterRole
+//
+// - ClusterRoleBinding
+//
+// - SecurityContextConstraint
+func createPrivilegedServiceAccount(ctx context.Context, log *logrus.Entry, name, cluster, usersAccount string, kubeActions adminactions.KubeActions) (func() error, error) {
+	serviceAcc := newServiceAccount(name, cluster)
+	log.Infof("Creating Service Account %s now", serviceAcc.GetName())
+	err := kubeActions.KubeCreateOrUpdate(ctx, serviceAcc)
+	if err != nil {
+		return nil, err
+	}
+
+	clusterRole := newClusterRole(usersAccount, cluster)
+	log.Infof("Creating Cluster Role %s now", clusterRole.GetName())
+	err = kubeActions.KubeCreateOrUpdate(ctx, clusterRole)
+	if err != nil {
+		return nil, err
+	}
+
+	crb := newClusterRoleBinding(name, cluster)
+	log.Infof("Creating Cluster Role Binding %s now", crb.GetName())
+	err = kubeActions.KubeCreateOrUpdate(ctx, crb)
+	if err != nil {
+		return nil, err
+	}
+
+	scc := newSecurityContextConstraint(name, cluster, usersAccount)
+	log.Infof("Creating Security Context Constraint %s now", name)
+	err = kubeActions.KubeCreateOrUpdate(ctx, scc)
+	if err != nil {
+		return nil, err
+	}
+
+	return func() error {
```

Comment:
@cadenmarchese This is a good point, the function won't execute if an error occurs in `createPrivilegedServiceAccount()`.
I'll reorder the cleanup function to execute here as well so that no dangling permissions are potentially left.